xmake/rules/c++/modules/modules_support/clang.lua

```diff
@@ -185,13 +185,13 @@ function generate_user_headerunits(target, batchcmds, headerunits, opt)
             outdir = path.join(cachedir, path.directory(headerunit.path))
         end
         if not os.isdir(outdir) then
-            os.mkdir(outdir)
+            batchcmds:mkdir(outdir)
```

Comment:
we can remove `if not os.isdir(outdir) then`, need only batchcmds:mkdir, it will handle it.

```lua
function _runcmd_mkdir(cmd, opt)
    local dir = cmd.dir
    if not opt.dryrun and not os.isdir(dir) then
        os.mkdir(dir)
    end
end
```